src/AppInstallerCommonCore/Downloader.cpp

```diff
@@ -121,6 +127,138 @@ namespace AppInstaller::Utility
+        wil::unique_hinternet hRequest(HttpOpenRequest(hConnect.get(), L"POST", L"msix/generate", L"HTTP/1.1", NULL, acceptTypes, INTERNET_FLAG_RELOAD | INTERNET_FLAG_SECURE, 0));
+        THROW_LAST_ERROR_IF_NULL_MSG(hRequest.get(), "HttpOpenRequest() failed.");
+        DWORD headerLength;
+        SIZETToDWord(header.length(), &headerLength);
+        DWORD bodyLength;
+        SIZETToDWord(body.length(), &bodyLength);
+        THROW_LAST_ERROR_IF_MSG(!HttpSendRequestA(hRequest.get(), header.c_str(), headerLength, (void*)body.c_str(), bodyLength), "HttpSendRequestA() failed");
+    
+        // Check http return status
+        DWORD requestStatus = 0;
+        DWORD cbRequestStatus = sizeof(requestStatus);  
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(hRequest.get(),
+        HTTP_QUERY_STATUS_CODE | HTTP_QUERY_FLAG_NUMBER,
+        &requestStatus,
+        &cbRequestStatus,
+        nullptr), "Query download request status failed.");
+
+        if (requestStatus != HTTP_STATUS_OK)
+        {
+            AICLI_LOG(Core, Error, << "Download request failed. Returned status: " << requestStatus);
+            THROW_HR_MSG(MAKE_HRESULT(SEVERITY_ERROR, FACILITY_HTTP, requestStatus), "Download request status is not success.");
+        }
+
+        AICLI_LOG(Core, Verbose, << "Download request status success.");
+
+        DWORD dwSize = 4096;
+
+        char lpOutBuffer[4096] = { 0 };
+
+        //Get publisher and package names
+        StringCchPrintfA((LPSTR)lpOutBuffer, dwSize, "pwabuilder-package-publisher");
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(hRequest.get(), HTTP_QUERY_CUSTOM,
+            lpOutBuffer, &dwSize, NULL), "Could not get publisher name.");
+  
+        publisher.append(lpOutBuffer);
+
+
+        StringCchPrintfA((LPSTR)lpOutBuffer, dwSize, "pwabuilder-package-name");
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(hRequest.get(), HTTP_QUERY_CUSTOM,
+            lpOutBuffer, &dwSize, NULL), "Could not get package name.");
+
+        name.append(std::move(lpOutBuffer));
+
+        LONGLONG contentLength = 0;
+        DWORD cbContentLength = sizeof(contentLength);
+
+        THROW_LAST_ERROR_IF_MSG(!HttpQueryInfoA(
+            hRequest.get(),
+            HTTP_QUERY_CONTENT_LENGTH | HTTP_QUERY_FLAG_NUMBER64,
+            &contentLength,
+            &cbContentLength,
+            nullptr), "Query download request failed.");
+        AICLI_LOG(Core, Verbose, << "Download size: " << contentLength);
+
+        const int bufferSize = 1024 * 1024; // 1MB
+        auto buffer = std::make_unique<BYTE[]>(bufferSize);
+
+        BOOL readSuccess = true;
+        DWORD bytesRead = 0;
+        LONGLONG bytesDownloaded = 0;
+
+        do
+        {
+            if (progress.IsCancelled())
+            {
+                AICLI_LOG(Core, Info, << "Download cancelled.");
+                return;
+            }
+
+            readSuccess = InternetReadFile(hRequest.get(), buffer.get(), bufferSize, &bytesRead);
+

```

Comment:
This file has mixed vertical whitespacing. Consider a more condensed style so that linebreaks clearly delineate logical chunks of functionality.